test/org/zaproxy/zap/extension/pscanrulesAlpha/UserControlledCookieScannerUnitTest.java

```diff
@@ -0,0 +1,157 @@
+		HttpMessage msg = new HttpMessage();
+		HttpRequestHeader requestHeader = new HttpRequestHeader();
+		try {
+			requestHeader.setURI(new URI("http://example.com/i.php", false));
+		} catch (URIException | NullPointerException e) {
+		}
+		requestHeader.setMethod(HttpRequestHeader.GET);
+
+		msg = new HttpMessage();
+		msg.setRequestHeader(requestHeader);
+		msg.getResponseHeader().setStatusCode(HttpStatusCode.OK);
+		msg.getResponseHeader().addHeader(HttpHeader.CONTENT_TYPE, "text/html");
+		return msg;
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseDoesntSetCookie() {
+		// Given
+		HttpMessage msg = createMessage();
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestHasNoGetParams() {
+		// Given
+		HttpMessage msg = createMessage();
+		// WHen
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfRequestParamsHaveNoValues() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=&name=", false));
+		msg.getResponseHeader().setHeader(HttpResponseHeader.SET_COOKIE, "Set-Cookie: aCookie=aValue; Secure");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfCookieHasNoValue() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=&name=fred", false));
+		msg.getResponseHeader().setHeader(HttpResponseHeader.SET_COOKIE, "Set-Cookie: aCookie=\"\"; Secure");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfCookieIsEmpty() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=&name=fred", false));
+		msg.getResponseHeader().setHeader(HttpResponseHeader.SET_COOKIE, "");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldRaiseAlertIfRequestParamsAppearAsCookieValue() throws URIException, NullPointerException {
+		// Given
+		HttpMessage msg = createMessage();
+		msg.getRequestHeader().setURI(new URI("http://example.com/i.php?place=&name=fred", false));
+		msg.getResponseHeader().setHeader(HttpResponseHeader.SET_COOKIE, "Set-Cookie: aCookie=fred; Secure");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(1));

```

Comment:
Could also assert the alert's param (same for following tests).

Reply:
Agreed